packages/go_router/lib/src/delegate.dart

```diff
@@ -42,22 +43,57 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
   /// Set to true to disable creating history entries on the web.
   final bool routerNeglect;
 
-  final GlobalKey<NavigatorState> _key = GlobalKey<NavigatorState>();
-
-  RouteMatchList _matches = RouteMatchList.empty();
+  RouteMatchList _matchList = RouteMatchList.empty();
   final Map<String, int> _pushCounts = <String, int>{};
+  final RouteConfiguration _configuration;
+
+  @override
+  Future<bool> popRoute() async {
+    // Iterate backwards through the RouteMatchList until seeing a GoRoute
+    // with a non-null parentNavigatorKey or a ShellRoute with a non-null parentNavigatorKey
+    // and pop from that Navigator instead of the root.
+    final int matchCount = _matchList.matches.length;
+    for (int i = matchCount - 1; i >= 0; i--) {
```

Comment:
flutter style guide suggest https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-over
```suggestion
    for (int i = matchCount - 1; i >= 0; i -= 1) {
```

Reply:
https://github.com/flutter/packages/pull/2453/commits/61ebdb088bd2909bb2a838b8e76ccbcea83327b9